src/Particle/ParticleContainer.cpp

```diff
@@ -14,21 +14,66 @@ namespace
 {
     using namespace amrex;
 
-    // Note - this function MUST be consistent with AMReX_Particle.H
-    Long unpack_id (uint64_t idcpu) {
-        Long r = 0;
+    py::object pack_ids (py::array_t<uint64_t> idcpus,
+                         py::array_t<amrex::Long> ids)
+    {
+        if (idcpus.ndim() != 1) {
+            throw std::runtime_error("Input should be 1-D NumPy array");
+        }
+
+        auto buf = idcpus.request();
+        auto buf2 = ids.request();
+        if (buf.size != buf2.size) throw std::runtime_error("sizes do not match!");
```

Comment:
Shall we enclose this in parenthesis just for consistency with the other one-line `if` conditions?
```suggestion
        if (buf.size != buf2.size) {
            throw std::runtime_error("sizes do not match!");
        }
```